src/Compilers/Core/Portable/CommandLine/AnalyzerConfig.SectionNameMatching.cs

```diff
@@ -125,8 +126,15 @@ internal static bool TryUnescapeSectionName(string sectionName, out string? esca
                 {
                     sb.Append(lexer.EatCurrentCharacter());
                 }
+                else
+                {
+                    // This contained some sort of actual token, which is disallowed in globalconfig sections; don't use it when trying to match.
+                    Debug.Fail("This section should have been filtered out in an earlier call below.");
+                    unescapedSectionName = null;
+                    return false;
+                }
             }
-            escapedSectionName = sb.ToString();
+            unescapedSectionName = sb.ToString();
```

Comment:
💡 If there is a majority case where `unescapedSectionName==sectionName`, it may be beneficial to share the string instance since both are held by `GlobalAnalyzerConfig`.

Reply:
No reason to check anymore, since we're not holding onto the original section anymore.